discovery_engine_core/core/src/storage.rs

```diff
@@ -225,4 +237,24 @@ pub mod models {
         pub(crate) size: u32,
         pub(crate) next_page: u32,
     }
+
+    /// Returns the context in which the reaction happened.
+    ///
+    /// In other words this returns the information needed
+    /// to continue processing the reaction based on the
+    /// same storage snapshot on which the reaction was stored.
+    ///
+    /// Storage implementations which do not use snapshots might
+    /// have less strict consistency guarantees.
+    pub(crate) enum ReactionContext {
+        Positive {
+            embedding: Embedding,
+            snippet: String,
+            title: String,
```

Comment:
the market field is added in #515, stack id is currently blocked on it not being in the db